test/e2e/node_feature_discovery_test.go

```diff
@@ -518,14 +532,17 @@ var _ = SIGDescribe("NFD master and worker", func() {
 							nfdv1alpha1.FeatureLabelNs + "/fake-fakefeature3":      "overridden",
 						},
 					}
-					Expect(waitForNfdNodeLabels(f.ClientSet, expectedLabels)).NotTo(HaveOccurred())
-
+					Expect(checkForNodeLabels(f.ClientSet,
+						expectedLabels, oldLabels,
+					)).NotTo(HaveOccurred())
 					By("Deleting NodeFeature object")
 					err = nfdClient.NfdV1alpha1().NodeFeatures(f.Namespace.Name).Delete(context.TODO(), nodeFeatures[0], metav1.DeleteOptions{})
 					Expect(err).NotTo(HaveOccurred())
 
 					By("Verifying node labels from NodeFeature object were removed")
-					Expect(waitForNfdNodeLabels(f.ClientSet, nil)).NotTo(HaveOccurred())
+					Expect(checkForNodeLabels(f.ClientSet,
+						map[string]k8sLabels{}, oldLabels,
```

Comment:
You could probably use just nil here?

```suggestion
						nil, oldLabels,
```